diff --git a/Library/Homebrew/extend/ENV/super.rb b/Library/Homebrew/extend/ENV/super.rb
index 5534cc8b0..26cbc93cf 100644
--- a/Library/Homebrew/extend/ENV/super.rb
+++ b/Library/Homebrew/extend/ENV/super.rb
@@ -55,6 +55,7 @@ module Superenv
     super
     send(compiler)
 
+    self["MACOSX_DEPLOYMENT_TARGET"] = "10.11"
     self["HOMEBREW_ENV"] = "super"
     self["MAKEFLAGS"] ||= "-j#{determine_make_jobs}"
     self["PATH"] = determine_path
diff --git a/Library/Homebrew/extend/os/mac/hardware.rb b/Library/Homebrew/extend/os/mac/hardware.rb
index 2b6e8de4e..0401dd6fd 100644
--- a/Library/Homebrew/extend/os/mac/hardware.rb
+++ b/Library/Homebrew/extend/os/mac/hardware.rb
@@ -5,6 +5,7 @@ module Hardware
   extend T::Sig
   sig { params(version: T.nilable(Version)).returns(Symbol) }
   def self.oldest_cpu(version = MacOS.version)
+    return :core2
     if CPU.arch == :arm64
       :arm_vortex_tempest
     # TODO: this cannot be re-enabled until either Rosetta 2 supports AVX
