diff --git a/Library/Homebrew/extend/ENV/super.rb b/Library/Homebrew/extend/ENV/super.rb
index 6be9069013..55ca7c6bf4 100644
--- a/Library/Homebrew/extend/ENV/super.rb
+++ b/Library/Homebrew/extend/ENV/super.rb
@@ -60,6 +60,7 @@ module Superenv
     super
     send(compiler)
 
+    self["MACOSX_DEPLOYMENT_TARGET"] = "12.0"
     self["HOMEBREW_ENV"] = "super"
     self["MAKEFLAGS"] ||= "-j#{determine_make_jobs}"
     self["RUSTFLAGS"] = Hardware.rustflags_target_cpu
